third_party/nvfuser/csrc/tensor_view.cpp

```diff
@@ -1305,10 +1305,10 @@ TensorView* TensorView::cacheAfter(c10::optional<LoadStoreOpType> cache_op) {
 
   // Get all the uses for this Tensorview
   TORCH_CHECK(
-      !isFusionOutput(),
+      !(uses().empty() && isFusionOutput()),
```

Comment:
is `isFusionOutput` necessary? If a tensor is not used, `cacheAfter` doesn't make any sense.

FYI, if a tensor is not used and is not a fusion output, it is usually not computed as it isn't live, except for when a tensor is an output of multi-output expressions. When only part of the output tensors are used, the unused outputs are still computed but their `uses()` should be empty.